src/commands/deploy.js

```diff
@@ -16,20 +16,263 @@ const SitesCreateCommand = require('./sites/create')
+      deployTimeout: flags.timeout * 1000 || 1.2e6,
+      syncFileLimit: 100,
+      branch: alias,
+    })
+  } catch (e) {
+    switch (true) {
+      case e.name === 'JSONHTTPError': {
+        warn(`JSONHTTPError: ${e.json.message} ${e.status}`)
+        warn(`\n${JSON.stringify(e, null, '  ')}\n`)
+        error(e)
+        return
+      }
+      case e.name === 'TextHTTPError': {
+        warn(`TextHTTPError: ${e.status}`)
+        warn(`\n${e}\n`)
+        error(e)
+        return
+      }
+      case e.message && e.message.includes('Invalid filename'): {
+        warn(e.message)
+        error(e)
+        return
+      }
+      default: {
+        warn(`\n${JSON.stringify(e, null, '  ')}\n`)
+        error(e)
+        return
+      }
+    }
+  }
+
+  const siteUrl = results.deploy.ssl_url || results.deploy.url
+  const deployUrl = get(results, 'deploy.deploy_ssl_url') || get(results, 'deploy.deploy_url')
+  const logsUrl = `${get(results, 'deploy.admin_url')}/deploys/${get(results, 'deploy.id')}`
+
+  return {
+    name: results.deploy.deployId,
+    siteId: results.deploy.site_id,
+    siteName: results.deploy.name,
+    deployId: results.deployId,
+    siteUrl,
+    deployUrl,
+    logsUrl,
+  }
+}
+
+const printResults = ({ flags, results, deployToProduction, log, logJson, exit }) => {
+  const msgData = {
+    'Logs': `${results.logsUrl}`,
+    'Unique Deploy URL': results.deployUrl,
+  }
+
+  if (deployToProduction) {
+    msgData['Website URL'] = results.siteUrl
+  } else {
+    delete msgData['Unique Deploy URL']
+    msgData['Website Draft URL'] = results.deployUrl
+  }
+
+  // Spacer
+  log()
+
+  // Json response for piping commands
+  if (flags.json) {
+    const jsonData = {
+      name: results.name,
+      site_id: results.site_id,
+      site_name: results.siteName,
+      deploy_id: results.deployId,
+      deploy_url: results.deployUrl,
+      logs: results.logsUrl,
+    }
+    if (deployToProduction) {
+      jsonData.url = results.siteUrl
+    }
+
+    logJson(jsonData)
+    exit(0)
+  } else {
+    log(prettyjson.render(msgData))
+
+    if (!deployToProduction) {
+      log()
+      log('If everything looks good on your draft URL, deploy it to your main site URL with the --prod flag.')
+      log(`${chalk.cyanBright.bold('netlify deploy --prod')}`)
+      log()
+    }
+  }
+}
+
 class DeployCommand extends Command {
   async run() {
     const { flags } = this.parse(DeployCommand)
+    const { log, logJson, warn, error, exit } = this
     const { api, site, config } = this.netlify
     const alias = flags.alias || flags.branch
 
     if (flags.branch) {
-      this.warn('--branch flag has been renamed to --alias and will be removed in future versions')
+      warn('--branch flag has been renamed to --alias and will be removed in future versions')
     }
 
     const deployToProduction = flags.prod
-    if (deployToProduction && alias) {
-      this.error(`--prod and --alias flags cannot be used at the same time`)
```

Comment:
Replaced this with https://github.com/netlify/cli/pull/1240/files#diff-e9001bf37e31e33db8a717da3ac59a69R474